### ISSUE TITLE
SQLType is incomplete

### ISSUE THREAD

**Issue by webloft** [mysql-d/mysql-native#225]
According to https://dev.mysql.com/doc/dev/connector-net/6.10/html/T_MySql_Data_MySqlClient_MySqlDbType.htm there are more types to support. 

BINARY is currently treaten as string.

**Comment by schveiguy** [comment]
This came up recently. I looked into it further, the reason it comes back as `string` is because that's what the server is reporting. A CHAR/VARCHAR with character set of BINARY.

However, I think it makes sense that the BINARY type should be mapped to `ubyte[]`.

I think this is the only thing that is missing, right?

**Comment by webloft** [comment]
Yeah, defintely since that padded string may not conform with a UTF-sequence. And VARBINARY could also be a non-UTF string, so both should be `ubyte[]`.

I would say GEOMETRY is the last missing type like POINT, POLYGON, LINESTRING, MULTILINESTRING, etc but I don't know how these types get transported. I read about WKB (Well-Known Binary) serialization format. If it comes in this format then there should be a proper mapping for it, I think. Or is it just text?

**Comment by schveiguy** [comment]
What type do you get back from the server for GEOMETRY?

**Comment by webloft** [comment]
Did a quick test with a script: If I put a POINT in it, a binary string is returned. Looks like that WKB format.
